packages/backend/src/inMemoryAnalytics.ts

```diff
@@ -0,0 +1,77 @@
+import { tableFromJSON, tableToIPC } from 'apache-arrow';
+import duckdb from 'duckdb';
+import { promisify as nodePromisify } from 'util';
+import Logger from './logging/logger';
+
+type InMemoryJSONRow = Record<string, unknown>;
+type InMemoryJSONTableMap = Record<string, InMemoryJSONRow[]>;
+
+export interface InMemoryDatabaseProvisioningOptions {
+    tables: InMemoryJSONTableMap;
+}
+
+/**
+ * Creates a new in-memory DuckDB instance, and exposes the instance itself
+ * alongside some convenience wrappers.
+ *
+ * The following is a potential alternative that includes an async version of the client,
+ * but for now we're opting for convenience + avoiding version lock from a third-party package:
+ *
+ * https://www.npmjs.com/package/duckdb-async
```

Comment:
This looks reliable, it's owned by `motherduck` the commercial entity that sponsors `duckdb` - let's do it and save our selves the promisifying